src/test/java/uk/gov/companieshouse/web/accounts/util/ValidationHelperTest.java

```diff
@@ -0,0 +1,233 @@
+import uk.gov.companieshouse.web.accounts.exception.MissingValidationMappingException;
+import uk.gov.companieshouse.web.accounts.validation.ValidationError;
+import uk.gov.companieshouse.web.accounts.validation.ValidationMapping;
+import uk.gov.companieshouse.web.accounts.validation.ValidationModel;
+
+@ExtendWith(MockitoExtension.class)
+public class ValidationHelperTest {
+    
+    private static final String BEAN_CLASS_NAME = MockValidationModel.class.getName();
+    private static final String RECURSIVE_BEAN_NAME = MockRecursiveValidationModel.class.getName();
+    private static final String PRIMITIVES_ONLY_BEAN_NAME = MockPrimitivesOnlyValidationModel.class.getName();
+    private static final String INVALID_BEAN_CLASS_NAME = MockValidationModel.class.getName() + "invalid";
+    
+    private static final String BASE_PATH = "test";
+    private static final String JSON_PATH = "json.path";
+    private static final String FIELD_PATH = "mockString";
+    private static final String MESSAGE_KEY = "invalid_character";
+    private static final String WEB_ERROR_MESSAGE = "validation.character.invalid";
+    private static final String ERROR = "error";
+    private static final Map<String, String> MESSAGE_ARGUMENTS = new HashMap<>();
+
+    @Mock
+    private ClassPathScanningCandidateComponentProvider mockProvider;
+    
+    @InjectMocks
+    private ValidationHelper helper;
+    
+    @BeforeAll
+    public static void setUpData() {
+        MESSAGE_ARGUMENTS.put(MESSAGE_KEY, ERROR);
+    }
+    
+    @Test
+    public void testSuccessfulScan() {
+        mockComponentScanning(BEAN_CLASS_NAME);
+        ValidationHelper.scanPackageForValidationMappings(mockProvider, BASE_PATH);
+    }
+    
+    @Test
+    public void testNoMappingsFound() {
+        assertThrows(IllegalStateException.class, () -> ValidationHelper.scanPackageForValidationMappings(mockProvider, BASE_PATH));
+    }
+    
+    @Test
+    public void testValidationMappingBeanNotFound() {
+        mockComponentScanning(INVALID_BEAN_CLASS_NAME);
+        assertThrows(IllegalStateException.class, () -> ValidationHelper.scanPackageForValidationMappings(mockProvider, BASE_PATH));
+    }
+    
+    @Test
+    public void testNoAnnotationsInValidationModel() {
+        mockComponentScanning(PRIMITIVES_ONLY_BEAN_NAME);
+        assertThrows(IllegalStateException.class, () -> ValidationHelper.scanPackageForValidationMappings(mockProvider, BASE_PATH));
+    }
+    
+    @Test
+    public void testScanMaxDepthReached() {
+        mockComponentScanning(RECURSIVE_BEAN_NAME);
+        assertThrows(IllegalStateException.class, () -> ValidationHelper.scanPackageForValidationMappings(mockProvider, BASE_PATH));
+    }
+
+    @Test
+    public void testGetValidationError() throws JsonProcessingException {
+        mockComponentScanning(BEAN_CLASS_NAME);
+        ValidationHelper.scanPackageForValidationMappings(mockProvider, BASE_PATH);
+        
+        List<ApiError> errors = createErrors(1);
```

Comment:
Don't need to assign a variable here, just use `createErrors(1)` in the method call below